python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -145,22 +160,24 @@ def _get_socket(self):
             self.logger.error("Impossible to connect - authentication failure ?")
             return None
 
-        # Socket is not opened anymore, try to reconnect
-        timeout = MQTTWrapper.TIMEOUT_RECONNECT_S
-        while timeout > 0:
+        # Socket is not opened anymore, try to reconnect for timeout if set
+        timeout = self.timeout
+        no_timeout = (timeout == 0)
+
+        while no_timeout or timeout > 0:
             try:
                 ret = self._client.reconnect()
                 if ret == mqtt.MQTT_ERR_SUCCESS:
                     break
             except Exception:
-                # Retry to connect in 1 sec up to timeout
+                # Retry to connect in 1 sec up to timeout if set
                 sleep(1)
                 timeout -= 1
                 self.logger.debug("Retrying to connect in 1 sec")
 
-        if timeout <= 0:
+        if not no_timeout and timeout <= 0:
```

Comment:
Similar thing here, isnt this the same as 

`if timeout < 0 :`

Reply:
Same reply, no :)

I definitely have to write it differently.